internalshared/configutil/userlockout.go

```diff
@@ -0,0 +1,182 @@
+				}
+			}
+		}
+		userLockoutsMap[userLockoutConfig.Type] = &userLockoutConfig
+	}
+
+	userLockoutsMap = setMissingUserLockoutValuesInMap(userLockoutsMap)
+	for _, userLockoutValues := range userLockoutsMap {
+		result.UserLockouts = append(result.UserLockouts, userLockoutValues)
+	}
+	return nil
+}
+
+// setUserLockoutValueAllInMap sets default user lockout values for key "all" (all auth methods)
+// for user lockout fields that are not configured using config file
+func setUserLockoutValueAllInMap(userLockoutAll *UserLockout) *UserLockout {
+	if userLockoutAll.Type == "" {
+		userLockoutAll.Type = "all"
+	}
+	if userLockoutAll.LockoutThresholdRaw == nil {
+		userLockoutAll.LockoutThreshold = UserLockoutThresholdDefault
+	}
+	if userLockoutAll.LockoutDurationRaw == nil {
+		userLockoutAll.LockoutDuration = UserLockoutDurationDefault
+	}
+	if userLockoutAll.LockoutCounterResetRaw == nil {
+		userLockoutAll.LockoutCounterReset = UserLockoutCounterResetDefault
+	}
+	if userLockoutAll.DisableLockoutRaw == nil {
+		userLockoutAll.DisableLockout = DisableUserLockoutDefault
+	}
+	return setNilValuesForRawUserLockoutFields(userLockoutAll)
+}
+
+// setDefaultUserLockoutValuesInMap sets missing user lockout fields for auth methods
+// with default values (from key "all") that are not configured using config file
+func setMissingUserLockoutValuesInMap(userLockoutsMap map[string]*UserLockout) map[string]*UserLockout {
+	// set values for "all" key with default values for "all" user lockout fields that are not configured
+	// the "all" key values will be used as default values for other auth methods
+	userLockoutAll, ok := userLockoutsMap["all"]
+	switch ok {
+	case true:
+		userLockoutsMap["all"] = setUserLockoutValueAllInMap(userLockoutAll)
+	default:
+		userLockoutsMap["all"] = setUserLockoutValueAllInMap(&UserLockout{})
+	}
+
+	for _, userLockoutAuth := range userLockoutsMap {
+		if userLockoutAuth.Type == "all" {
+			continue
+		}
+		// set missing values
+		if userLockoutAuth.LockoutThresholdRaw == nil {
+			userLockoutAuth.LockoutThreshold = userLockoutsMap["all"].LockoutThreshold
+		}
+		if userLockoutAuth.LockoutDurationRaw == nil {
+			userLockoutAuth.LockoutDuration = userLockoutsMap["all"].LockoutDuration
+		}
+		if userLockoutAuth.LockoutCounterResetRaw == nil {
+			userLockoutAuth.LockoutCounterReset = userLockoutsMap["all"].LockoutCounterReset
+		}
+		if userLockoutAuth.DisableLockoutRaw == nil {
+			userLockoutAuth.DisableLockout = userLockoutsMap["all"].DisableLockout
+		}
+		userLockoutAuth = setNilValuesForRawUserLockoutFields(userLockoutAuth)
+		userLockoutsMap[userLockoutAuth.Type] = userLockoutAuth
+	}
+	return userLockoutsMap
+}
+
+// setNilValuesForRawUserLockoutFields sets nil values for user lockout Raw fields
+func setNilValuesForRawUserLockoutFields(userLockout *UserLockout) *UserLockout {
+	if userLockout.LockoutThresholdRaw != nil {
```

Comment:
Would you think that these if statements are required? I think we could simply just assign nil to all of them without checking.